src/main/java/qupath/AnnotationExchangeExtension/ExportAnnotationServiceJSONPlugin.java

```diff
@@ -203,25 +203,29 @@ private boolean writeJSONAnnotations(File outputFile, ImageData<BufferedImage> i
                     pathProperties.addProperty("closed", true);
                     JsonArray fillColour = new JsonArray();
 
-                    if (annotation.getPathClass() == null) {
-                        fillColour.add(1);
-                        fillColour.add(0);
-                        fillColour.add(0);
-                        fillColour.add(0.5);
-                    } else {
-                        int annotationRGB = annotation.getPathClass().getColor();
-                        fillColour.add((double) (ColorTools.red(annotationRGB)) / 255.0);
-                        fillColour.add((double) (ColorTools.green(annotationRGB)) / 255.0);
-                        fillColour.add((double) (ColorTools.blue(annotationRGB)) / 255.0);
-                        fillColour.add(0.5);
-                    }
-
-                    pathProperties.add("fillColor", fillColour);
                     JsonArray strokeColor = new JsonArray();
-                    strokeColor.add(0);
-                    strokeColor.add(0);
-                    strokeColor.add(0);
+
+                    /**
+                     * PathObject.color is null by default, thus the color of the annotation needs to be manually set if
+                     * the user used the default color of RGB(255, 0, 0) (i.e. Red)
+                     */
+                    final int annotationRGB = annotation.getColorRGB() != null
+                        ? annotation.getColorRGB()
+                        : 16711680;
+                    final double redValue = (double) (ColorTools.red(annotationRGB)) / 255.0;
+                    final double greenValue = (double) (ColorTools.green(annotationRGB)) / 255.0;
+                    final double blueValue = (double) (ColorTools.blue(annotationRGB)) / 255.0;
+                    fillColour.add(redValue);
```

Comment:
could this be written more concisely?
```
JsonArray fillColor = new JsonArray([redValue, greenValue, blueValue, 0.5]);
```
or something like that? (pardon ny zero knowledge of java)

Reply:
also similarly with strokeColor below?
```
JsonArray strokeColor = new JsonArray([redValue, greenValue, blueValue]);
```

Reply:
> could this be written more concisely?
> 
> `JsonArray fillColor = new JsonArray([redValue, greenValue, blueValue, 0.5]);`
> or something like that? (pardon ny zero knowledge of java)

I wish =( In my experience, Java is very picky with instantiating Arrays/HashMaps with values, but I think in this particular case, `JsonArray` doesn't have an overloaded constructor that'll allow an `Array` as a parameter, even in the latest version of `com.google.gson`:

https://javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/JsonArray.html

As such, when I try to compile the extension:

![image](https://user-images.githubusercontent.com/16601729/70456145-6d3c3180-1a62-11ea-9004-eba9d031045a.png)

Reply:
:S that's unfortunate, I had googled it and found this https://stleary.github.io/JSON-java/org/json/JSONArray.html (wasn't sure where to find java docs) but looks like that's something else then

Reply:
Ahh, looking at your link, it looks like that's a different package than we're using for this extension.

Your link is for the package `org.json.JSONArray`, whereas this extension was built using `com.google.code.JSONArray`.

May be worth switching to `org.json.JSONArray` in the future to make the code more compact.